src/main/java/at/fhv/td/rmi/BuyTicketImpl.java

```diff
@@ -0,0 +1,78 @@
+package at.fhv.td.rmi;
+
+import at.fhv.td.domain.Ticket;
+import at.fhv.td.domain.assembler.TicketAssembler;
+import at.fhv.td.dto.TicketAnswerDTO;
+import at.fhv.td.dto.TicketDTO;
+import at.fhv.td.persistence.broker.TicketBroker;
+import at.fhv.td.rmi.interfaces.IBuyTicket;
+import at.fhv.td.rmi.interfaces.ITicketDTO;
+
+import java.rmi.RemoteException;
+import java.rmi.server.UnicastRemoteObject;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class BuyTicketImpl extends UnicastRemoteObject implements IBuyTicket {
+    private TicketAnswerDTO _ticketAnswer;
+
+    public BuyTicketImpl() throws RemoteException {
+        super();
+        _ticketAnswer = new TicketAnswerDTO();
+    }
+
+    @Override
+    public TicketAnswerDTO buyTicket(ITicketDTO ticketDto, Map<Integer, Integer[]> seatPlaceReservations) throws RemoteException {
+        LinkedList<Ticket> tickets;
+        _ticketAnswer = new TicketAnswerDTO();
+
+        tickets = createTickets(ticketDto, seatPlaceReservations, 1);
+
+        if (_ticketAnswer.getMessage() == null) {
+            _ticketAnswer.setMessage("Tickets successfully bought");
+            _ticketAnswer.setTickets(tickets.stream().map(TicketAssembler::toTicketDTO).toArray(TicketDTO[]::new));
+            tickets.forEach(ticket -> TicketBroker.getInstance().save(ticket));
```

Comment:
Ich glaube, dass die meiste Logik hier in die Applikationsschicht gehört 🤔  In der RMI Schicht sollte meiner Meinung nach nicht die Kauflogik abgebildet sein 🤔 
Auf die Persistenzschicht sollte aus der RMI Schicht auch nicht direkt zugegriffen werden denke ich ...

Vlt müssen wir darüber nochmal reden ... bin mir selbst nicht sicher.

Reply:
Stimmt, nochmal drüber reden. Ich sehe das RMI eher der Applikationsschicht angehörig, aber du kannst mich auch einfach überzeugen 😉